psycopg/psycopg/pq/_pq_ctypes.py

```diff
@@ -28,8 +28,8 @@ class FILE(Structure):
 FILE_ptr = POINTER(FILE)
 
 if sys.platform == "linux":
-    libcname = ctypes.util.find_library("c")
-    assert libcname
+    # find_library("c") does not work on musl, fall back to libc.so instead
+    libcname = ctypes.util.find_library("c") or "libc.so"
```

Comment:
Because the `libcname` is the normal and expected path (so much that there was an assert to enforce that), I would rather make it clear why there are cases where that's not the case.

```suggestion
    libcname = ctypes.util.find_library("c")
    if not cname:
        # Likely this is musl linux, see bugs ...
        # [maybe even add a check to verify we are on musl]
        libcname = "libc.so"
```

Reply:
I'm afraid that detecting musl would be non-trivial since musl upstream is of pretty strong opinion that people shouldn't be checking what libc is used and therefore do not provide any explicit way of checking for musl.